tool/subprocess_launcher.dart

```diff
@@ -14,11 +12,11 @@ import 'package:path/path.dart' as p;
 /// Keeps track of coverage data automatically for any processes run by this
 /// [CoverageSubprocessLauncher].  Requires that these be dart processes.
 class CoverageSubprocessLauncher extends SubprocessLauncher {
-  CoverageSubprocessLauncher(String context, [Map<String, String> environment])
-      : super(context, environment) {
-    environment ??= {};
-    environment['DARTDOC_COVERAGE_DATA'] = tempDir.path;
-  }
+  CoverageSubprocessLauncher(String context, [Map<String, String>? environment])
+      : super(
+            context,
+            (environment ?? {})
+              ..addAll({'DARTDOC_COVERAGE_DATA': tempDir.path}));
```

Comment:
no need for this PR, but I _think_ this is legal, and maybe more readable?

```dart
      : super(
            context,
            {
              ...?environment,
              'DARTDOC_COVERAGE_DATA': tempDir.path,
            }));
```

It's not... great 😝 , maybe a trailing comma here or there would improve it.